src/clr-angular/utils/animations/expandable/expandable.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Component, ElementRef, HostBinding, HostListener, Input } from '@angular/core';
+
+import { animate, style, transition, trigger } from '@angular/animations';
+import { DomAdapter } from '../../dom-adapter/dom-adapter';
+
+@Component({
+  selector: 'clr-expandable',
+  template: `
+    <ng-content></ng-content>
+  `,
+  styles: [
+    `
+    :host {
+      display: block;
+      overflow: hidden;
+    }
+  `,
+  ],
+  host: {
+    class: 'expandable',
+  },
+  animations: [
+    trigger('expandAnimation', [
+      transition('void => *', []),
+      transition('* => *', [
```

Comment:
I still don't see why you need `transition('void => *', []),`. 
Also, this transition `'* => *'` is overriding `'void => *'` because an asterisk * or wildcard matches any animation state. So the first transition is not even being used.

Reply:
Description:
We need the initial transition to avoid animating when we initialize a component as expanded.
Reproduction:
1. Go to https://stackblitz.com/edit/clarity-light-theme-v1-0-bt6xj9?file=src/app/expandable.component.ts
2. See commented transition on line 29. Uncomment it.
3. Observe that the component is rendered without animation.
4. Comment it.
5. Observe that now animation is visible when rendering the component anew.

Expected behavior:
No animation on initialization.

Hope that helps :)

Reply:
I have a similar empty transition in the stepper to prevent the animation animation on initialization.

Reply:
I see. I made the wrong assumption that the second transition would override the first one. Also, I wasn't completely sure whether it's a good idea or not to have the logic of escaping the first animation run in its metadata. We have another way of doing it, which is disabling the animation on initialization using: https://angular.io/api/animations/trigger#disabling-animations. But this is definitely not a blocker.